Mathlib/Order/Bounds/Basic.lean

```diff
@@ -1702,17 +1705,31 @@ The dual notion
 
 does not appear to play a significant role in the literature, so is omitted here.
 -/
-def ScottContinuous (f : α → β) : Prop :=
-  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a⦄, IsLUB d a → IsLUB (f '' d) (f a)
-#align scott_continuous ScottContinuous
+def DScottContinuous (D : Set (Set α)) (f : α → β) : Prop :=
+  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → d ∈ D → ∀ ⦃a⦄, IsLUB d a → IsLUB (f '' d) (f a)
```

Comment:
Shouldn't this be called `ScottContinuousOn`? or is "D-Scott continuity" a standard term?

Reply:
I made up the name `DScottContinuous`. I think most authors use either the chain definition or the directed set definition. Mathlib is a bit unusual in discussing both at once. Perhaps it should be `ScottContinuous'`?

Reply:
I mean compare to `Monotone` vs `MonotoneOn`. I think it makes a lot of sense for this to `ScottContinuousOn`. Also I'd suggest moving the `d ∈ D` binder to the front to match other such "restricted predicates".
```suggestion
def DScottContinuous (D : Set (Set α)) (f : α → β) : Prop :=
  ∀ ⦃d : Set α⦄, d ∈ D → d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a⦄, IsLUB d a → IsLUB (f '' d) (f a)
```

Reply:
That's a better name! Thanks.